go/obscuronode/enclave/state.go

```diff
@@ -273,19 +287,29 @@ func calculateBlockState(
 }
 
 // Todo - this has to be implemented differently based on how we define the ObsERC20
-func rollupPostProcessingWithdrawals(newHeadRollup *core.Rollup, state vm.StateDB, header *nodecommon.Header) []nodecommon.Withdrawal {
+func (e *enclaveImpl) rollupPostProcessingWithdrawals(newHeadRollup *core.Rollup, state *state.StateDB, receipts map[common.Hash]*types.Receipt) []nodecommon.Withdrawal {
 	w := make([]nodecommon.Withdrawal, 0)
-	withdrawalTxs := withdrawals(state, header.Hash())
 	// go through each transaction and check if the withdrawal was processed correctly
 	for i, t := range newHeadRollup.Transactions {
-		txData := core.TxData(&newHeadRollup.Transactions[i])
-		if txData.Type == core.WithdrawalTx && contains(withdrawalTxs, t.Hash()) {
-			w = append(w, nodecommon.Withdrawal{
-				Amount:  txData.Amount,
-				Address: txData.From,
-			})
+		found, address, amount := erc20contractlib.DecodeTransferTx(t)
+
+		if found && *t.To() == evm.Erc20ContractAddress && *address == evm.WithdrawalAddress {
+			receipt := receipts[t.Hash()]
+			if receipt != nil && receipt.Status == 1 {
+				signer := types.NewLondonSigner(big.NewInt(e.chainID))
+				from, err1 := types.Sender(signer, &newHeadRollup.Transactions[i])
```

Comment:
Nitpick but did you mean to leave this with the 1 on the end? Just a little bit distracting for the reader maybe